chain-abci/src/app/mod.rs

```diff
@@ -164,11 +164,33 @@ impl<T: EnclaveProxy> abci::Application for ChainNodeApp<T> {
                     .get(&validator_address)
                     .expect("Validator not found in liveness tracker")
                     .address();
-                self.jail_account(account_address)
-                    .expect("Unable to jail account in begin block");
+
+                accounts_to_jail.push(account_address);
             }
         }
 
+        let missed_block_threshold = self
+            .last_state
+            .as_ref()
+            .unwrap()
+            .jailing_config
+            .missed_block_threshold;
+
+        accounts_to_jail.extend(
+            self.last_state
+                .as_ref()
+                .unwrap()
+                .validator_liveness
```

Comment:
I'm thinking that this validator liveness tracking could be some struct wrapping the underlying map + containing the threshold -- it could then keep track of non-live validators after each `update_validator_liveness`... so here, it could directly return the non-live validators (instead of iterating over all validators and checking them)